src/utils/url.ts

```diff
@@ -0,0 +1,96 @@
+import lzString from 'lz-string';
+
+export interface Snippet {
+  group: string;
+  name: string;
+  code: string;
+}
+
+export type Snippets = Snippet[];
```

Comment:
Perhaps these `Snippet` types should be inside `types.d.ts`. Doesn't feel right that they're alongside URL-related stuff.

Reply:
makes sense, moved